packages/typescript-client/src/client.ts

```diff
@@ -207,53 +212,60 @@ export class ShapeStream {
       }
 
       try {
-        await this.fetchClient(url.toString(), { signal })
-          .then(async (response) => {
-            if (!response.ok) {
-              throw await FetchError.fromResponse(response, url.toString())
-            }
+        const response = await this.fetchClient(url.toString(), { signal })
 
-            const { headers, status } = response
-            const shapeId = headers.get(`X-Electric-Shape-Id`)
-            if (shapeId) {
-              this.shapeId = shapeId
-            }
+        if (!response.ok) {
+          throw await FetchError.fromResponse(response, url.toString())
+        }
 
-            const lastOffset = headers.get(`X-Electric-Chunk-Last-Offset`)
-            if (lastOffset) {
-              this.lastOffset = lastOffset as Offset
-            }
+        const { headers, status } = response
+        const shapeId = headers.get(`X-Electric-Shape-Id`)
+        if (shapeId) {
+          this.shapeId = shapeId
+        }
+
+        const lastOffset = headers.get(`X-Electric-Chunk-Last-Offset`)
+        if (lastOffset) {
+          this.lastOffset = lastOffset as Offset
+        }
 
-            attempt = 0
+        const schemaHeader = headers.get(`X-Electric-Schema`)!
+        this.schema = schemaHeader ? JSON.parse(schemaHeader) : {}
 
-            if (status === 204) {
-              return []
-            }
+        attempt = 0
 
-            return response.json() as Promise<Message[]>
-          })
-          .then((batch: Message[]) => {
-            // Update isUpToDate & lastOffset
-            if (batch.length > 0) {
-              const lastMessages = batch.slice(-2)
+        const messages = status === 204 ? `[]` : await response.text()
+        const batch = JSON.parse(messages, (key, value) => {
+          if (key === `value`) {
+            // Parse the row values
+            const row = value as Record<string, Value>
```

Comment:
this doesn't work correctly when there is a column named `value` - it's a good idea to actually add a test with a column named that. We prob need to `typeof value == 'object'` here

Reply:
Indeed, great catch!